src/components/Application/Navigation.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import { Button, Segment, Sticky } from 'semantic-ui-react'
+import {
+  MethodRevalidate,
+  MethodToCallProps,
+  SectionAndPage,
+  SectionDetails,
+  SectionsStructureNEW,
+} from '../../utils/types'
+import strings from '../../utils/constants'
+import { useRouter } from '../../utils/hooks/useRouter'
+
+interface NavigationProps {
+  current: SectionAndPage
+  sections: SectionsStructureNEW
+  serialNumber: string
+  requestRevalidation: MethodRevalidate
+}
+
+const Navigation: React.FC<NavigationProps> = ({
+  current,
+  sections,
+  serialNumber,
+  requestRevalidation,
+}) => {
+  const { push } = useRouter()
+
+  const currentSectionDetails = sections[current.sectionCode].details
+
+  const nextSection = (): SectionDetails | null => {
+    const nextSections = Object.values(sections)
+      .filter(({ details: { index } }) => index > currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => aIndex - bIndex)
+    return nextSections.length > 0 ? nextSections[0].details : null
+  }
+
+  const previousSection = (): SectionDetails | null => {
+    const previousSections = Object.values(sections)
+      .filter(({ details: { index } }) => index < currentSectionDetails.index)
+      .sort(({ details: { index: aIndex } }, { details: { index: bIndex } }) => bIndex - aIndex)
+    return previousSections.length > 0 ? previousSections[0].details : null
+  }
+
+  const isFirstPage = current.pageNumber - 1 === 0 && previousSection() == null
+  const isLastPage =
+    current.pageNumber + 1 > currentSectionDetails.totalPages && nextSection() == null
+
+  const getPreviousSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber > 1) return { sectionCode, pageNumber: pageNumber - 1 }
+    return {
+      sectionCode: (previousSection() as SectionDetails).code,
+      pageNumber: (previousSection() as SectionDetails).totalPages,
+    }
+  }
+
+  const getNextSectionPage = (): SectionAndPage => {
+    const { sectionCode, pageNumber } = current
+    if (pageNumber < currentSectionDetails.totalPages)
+      return { sectionCode, pageNumber: pageNumber + 1 }
+    return {
+      sectionCode: (nextSection() as SectionDetails).code,
+      pageNumber: 1,
+    }
+  }
```

Comment:
I feel like these four methods could be combined into a single generic method that can go backwards and forwards and the input parameter specifies by how much. 

To go forward a page, you'd call: `const nextSectionPage = getSectionPage(1)`
And to go back you'd call: `const prevSectionPage = getSectionPage(-1)`

And the `getSectionPage` method would basically do:
- flatten all pages to a single, sorted array of SectionAndPage objects (i.e. has sectionCode and pageNum)
- infer the index in this array that the current page is
- Add the input param to the currentPageIndex -- if greater than .length, set newIndex = .length; if less than 0, newIndex = 0.
- return the sectionPage details of the object at the newIndex (code and page num)

(This would also work for jumping multiple pages, not that we need it, but: `getSectionPage(3)` for example

Reply:
Hmmm I see your point, but really think it's ok since it's working correctly now and easy to read. And because no other use for this inside this component - jumping page for example. If required in the future we can change to that.

Reply:
Sure, I'm not a fan of all the duplicated code -- but @andreievg 's approved and it works so carry on 👍

Reply:
Yea, me neither, I would prefer to combine usually, and I see this solution as a possible one. I just think the very generic option wouldn't be as good since the Previous should get any validation at all

Reply:
Sorry, couldn't resist having a quick play to see what the difference is. Check out branch `350-go-to-page-handler` (branched off this one) to see the alternative. Feel free to ignore, just wanted to see how it compared. 

Otherwise, merge away!